### ISSUE TITLE
Suggestion: Use commands instead of &str to execute commands

### ISSUE THREAD

**Issue by Raj2032** [gluesql/gluesql#610]
Hey guys, I have been using gluesql for a while and I wanted to suggest improvements to it.

In your example:
```rust
use gluesql::prelude::*;

fn main() {
    let storage = SledStorage::new("data/doc-db").unwrap();
    let mut glue = Glue::new(storage);
    let sqls = vec![
        "DROP TABLE IF EXISTS Glue;",
        "CREATE TABLE Glue (id INTEGER);",
        "INSERT INTO Glue VALUES (100);",
        "INSERT INTO Glue VALUES (200);",
        "SELECT * FROM Glue WHERE id > 100;",
    ];

    for sql in sqls {
        let output = glue.execute(sql).unwrap();
        println!("{:?}", output)
    }
}
```

While this is great for basic stuff, I wanted to suggest a more cleaner alternative.

Here is my suggestion to use commands to execute stuff such as `select` etc etc. Here is my example (this is pseudo code):
```rust
glue.drop_table(["Glue"]);
glue.create_table("Glue", vec![id: INTEGER]); // Takes two arguments, one is for the table name, the other is for column names
glue.insert("Glue", vec!(100);
glue.insert("Glue", vect!(200);
glue.select("Glue", ["*"], [id > 100, id == 100]; // table_name, columns, condition(s)
```

Unless if there was a way to do this already?

**Comment by panarch** [comment]
Currently, the only interface you can use is SQL text.
However, there is a good news.
We are planning to add GlueSQL's own query builder so you will be able to use .. not soon but not too late.
Until this year, you will be able to see some visible working example of the new query builder.

**Comment by Raj2032** [comment]
@panarch Hey mate thanks for your response :)

>We are planning to add GlueSQL's own query builder so you will be able to use .. not soon but not too late.

Sorry mate but what is a query builder? Is it similar to the example I have provided?

Can I have a pseudo code example please?

**Comment by panarch** [comment]
@Raj2032 Sure,
[`sea-query`](https://github.com/SeaQL/sea-query) would be a good example.
You can expect similar feature like SeaQuery.
Differences are... SeaQuery is for supporting multiple SQL databases using same query interface.
In the case of the query builder which GlueSQL plans to make is only for us, GlueSQL.
It is designed to use full functionality of what GlueSQL provides.

Pros - Query builder can use full feature of GlueSQL, so there would be no issue to use gluesql-only-specific features using query builder.

Cons - Not compatible with other databases, GlueSQL query builder only works for GlueSQL.

**Comment by Raj2032** [comment]
@panarch If I understood you correctly, the gluesql query would be similar to sea:

```rust
assert_eq!(
    Query::select()
        .column(Glyph::Image)
        .from(Glyph::Table)
        .and_where(Expr::col(Glyph::Image).like("A"))
        .and_where(Expr::col(Glyph::Id).is_in(vec![1, 2, 3]))
        .build(PostgresQueryBuilder),
    (
        r#"SELECT "image" FROM "glyph" WHERE "image" LIKE $1 AND "id" IN ($2, $3, $4)"#
            .to_owned(),
        Values(vec![
            Value::String(Some(Box::new("A".to_owned()))),
            Value::Int(Some(1)),
            Value::Int(Some(2)),
            Value::Int(Some(3))
        ])
    )
);
```

**Comment by panarch** [comment]
Yes similar but method names and detailed usage would be quite different.
I'll share the draft in the public soon
also thanks for the question. this helps us to decide priority between issues.
